src/dash-playlist-loader.js

```diff
@@ -696,45 +710,63 @@ export default class DashPlaylistLoader extends EventTarget {
       this.masterPlaylistLoader_.srcUrl = location;
     }
 
-    // if the minimumUpdatePeriod was changed, update the minimumUpdatePeriodTimeout_
-    if (!oldMaster || (newMaster && oldMaster.minimumUpdatePeriod !== newMaster.minimumUpdatePeriod)) {
+    if (!oldMaster || (newMaster && newMaster.minimumUpdatePeriod !== oldMaster.minimumUpdatePeriod)) {
       this.updateMinimumUpdatePeriodTimeout_();
     }
 
     return Boolean(newMaster);
   }
 
   updateMinimumUpdatePeriodTimeout_() {
-    // Clear existing timeout
-    window.clearTimeout(this.minimumUpdatePeriodTimeout_);
+    const mpl = this.masterPlaylistLoader_;
 
-    const createMUPTimeout = (mup) => {
-      this.minimumUpdatePeriodTimeout_ = window.setTimeout(() => {
-        this.trigger('minimumUpdatePeriod');
-        createMUPTimeout(mup);
-      }, mup);
-    };
+    // cancel any pending creation of mup on media
+    // a new one will be added if needed.
+    if (mpl.createMupOnMedia_) {
+      mpl.off('loadedmetadata', mpl.createMupOnMedia_);
+      mpl.createMupOnMedia_ = null;
+    }
 
-    const minimumUpdatePeriod = this.masterPlaylistLoader_.master && this.masterPlaylistLoader_.master.minimumUpdatePeriod;
+    // clear any pending timeouts, master was just updated
+    // so if minimumUpdatePeriodTimeout_ is still valid
```

Comment:
I'm not sure if I understand this comment. Is it describing the scenario that, since the master playlist was just updated, this loader needs to check to see if the master playlist changed its minimum update period, and therefore should update its minimum update period timeout?

Reply:
Will clarify this comment better.